### PR TITLE
Remove enumerator in map.find

### PR DESCRIPTION
**Pull request by forki**
Now that we have new map optimized we should reevalute the usages. Here we can remove one layer of enumerators

### CODE REVIEW

File: src/fsharp/FSharp.Core/map.fs [hunk 1]
```diff
@@ -670,7 +670,8 @@ type Map<[<EqualityConditionalOn>]'Key, [<EqualityConditionalOn; ComparisonCondi
                 let m1 = e1.MoveNext() 
                 let m2 = e2.MoveNext()
                 (m1 = m2) && (not m1 || 
-                                 (let e1c, e2c = e1.Current, e2.Current
+                                 (let e1c = e1.Current
```
Comment (TIHan) [inline src/fsharp/FSharp.Core/map.fs:673]: Is this is a formatting change only? Both versions will be exactly the same. No tuple will be actually be created.

File: src/fsharp/FSharp.Core/map.fs [hunk 2]
```diff
@@ -540,7 +540,7 @@ type Map<[<EqualityConditionalOn>]'Key, [<EqualityConditionalOn; ComparisonCondi
     member __.OnDeserialized(context: System.Runtime.Serialization.StreamingContext) =
         ignore context
         comparer <- LanguagePrimitives.FastGenericComparer<'Key>
-        tree <- serializedData |> Array.map (fun (KeyValue(k, v)) -> (k, v)) |> MapTree.ofArray comparer
```
Comment (TIHan) [inline src/fsharp/FSharp.Core/map.fs:543]: Is this is a formatting change only?

**Review by TIHan — APPROVED** [review]
Removing `toSeq` in `findKey` and `tryFindKey` is a good change.

Just a few comments. Also, a very simple sentence describing the change in the description would be appreciated.

File: src/fsharp/FSharp.Core/map.fs (hunk 1 above)
Comment (forki) [reply src/fsharp/FSharp.Core/map.fs:673]: yes I think it reduces confusion here.

Comment (TIHan) [reply src/fsharp/FSharp.Core/map.fs:673]: Ok, that's fine.

**Review by TIHan — APPROVED** [review]

**Pull request merged**